### DISCUSSION TITLE
How to use SSG/SSR with a remote API (not hosted on next API routes)?

### DISCUSSION THREAD

**Discussion by sinclairnick** [trpc/trpc#4010] [Q&A] [answered]
I'm wanting to implement SSG in my NextJS app, with a remote API.

According to the SSG docs, you can call `createProxySSGHelpers` which just call the methods directly without an HTTP request. My tRPC server is not located in the same environment as the `getStaticProps` execution context, so this obviously won't work. Instead I just require the SSG helpers using an HTTP link.

I'm well versed in React Query, where I would typically do something like

```ts
const client = new QueryClient()

// Fetch data etc.

return {
  props: { dehydratedState: dehydrate(client) }
}
```

I've seen the docs regarding [React](https://trpc.io/docs/react#3-add-trpc-providers) which mentions the use of the `trpc.Provider` and a custom `QueryProvider`, which could possibly work here?

Another angle is to perhaps use the `createTRPCProxyClient` and mix that with `trpcState` via 
```tsx
queryClient.prefetchQuery(
  getQueryKey(...),
  trpcServer.some.sort.of.query(...)
)

return {
  props: {trpcState: dehydrate(queryClient)}
}
```

In any case, it's not clear how to achieve this behaviour, and it was a little bit odd to me that the docs only mention the tightly-coupled Prisma colocated case.

I was hoping for some advice on what the canonical method for achieving SSG for NextJS apps which call external tRPC APIs?

**Comment by KATT** [comment, marked as answer]
Looks like you want a `createProxySSGHelpers()`-equivalent that works with a client that does actual HTTP-calls. Feel free to open a feature request.

**Reply by sinclairnick** [reply]
Will do, thanks
